zerver/models/users.py

```diff
@@ -936,6 +946,10 @@ def get_user_by_delivery_email(email: str, realm: "Realm") -> UserProfile:
         "realm",
         "realm__can_access_all_users_group",
         "realm__can_access_all_users_group__named_user_group",
+        "realm__direct_message_initiator_group",
+        "realm__direct_message_initiator_group__named_user_group",
+        "realm__direct_message_permission_group",
+        "realm__direct_message_permission_group__named_user_group",
```

Comment:
I don't think we need to have every query fetching a user include fetching these tables.

@sahil839 FYI, probably this reflects a missing set of comments as to why we need specifically `realm__can_access_all_users_group` when fetching users.

Reply:
The reasons for fetching `can_access_all_users_group` is different for different cases. Like `get_user_by_delivery_email` and `maybe_get_user_profile_by_api_key` are called when setting the `user_profile` object passed in `views` functions. Others are mostly when messages are send by bots or webhooks and also when checking access to a user for a couple of endpoints.

Will mentioning one important reason where we actually do not want one extra query be enough?

Reply:
@Vector73 is this still necessary? I think b44d30add7297a7ad54c0ce7bfffacf2bdfd4b5b should have obviated the need to prefetch these items here.

Reply:
Some tests are failing if I remove these.

Reply:
We would still need to fetch these fields to not increase the query count in the code for checking message sending permission. So, this would still be needed in functions which set `user_profille` object which is passed to the views.
And `get_user_by_delivery_email` is used to fetch users in `example_user` in tests, which fails the test if we do not prefetch these.

Reply:
OK, we can try to manage this in a follow-up.